### DISCUSSION TITLE
MinIO list objects caching: bloom filter settings

### DISCUSSION THREAD

**Discussion by charlesyc** [minio/minio#12285] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
This is actually not a feature request but a question regarding to MinIO code.
From [here](https://github.com/minio/minio/blob/5c0a7189c7633374924b219fa48ff59cec253539/cmd/data-update-tracker.go#L45), we can see that false positive rate of the bloom filter for listing objects is set to 0.99.
But in [the PR of this change](https://github.com/minio/minio/pull/9208), said that the false positive rate is meant to be set to 0.01.
I have verified that with larger false positive rate, the bloom filter package (github.com/willf/bloom) will give larger m (maximum size) and k (the number of hashing functions):
given: item = 10000000, fp = 0.99, it produces m = 209185, k = 1
given: item = 10000000, fp = 0.01, it produces m = 95850584, k = 7

**Describe the solution you'd like**
Maybe we need code change.
But with larger m and k, the serialized bloom filter will also be larger so we need to balance here.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by charlesyc** [comment]
I hope I understand this right..😂
Maybe @klauspost could give some comments here..

**Comment by charlesyc** [comment]
Could anyone who is familiar with bloom filter give some comments here?

**Comment by klauspost** [comment, marked as answer]
@charlesyc Updates in https://github.com/minio/minio/pull/12305
